docs/common/agent-builder/_behavior-trees.md

```diff
@@ -0,0 +1,26 @@
+A behavior tree is a decision-making model used to determine which actions an agent (such as a character in a video game) should take in different situations.
+You can think of it as a flow chart that the AI thinks through, choosing appropriate actions based on what it knows about itself and its environment.
+Behavior trees are composed of nodes organized in a hierarchical structure that start at a root node and branch out to form different paths.
+Each path represents a line of decision-making that results in one or more actions for the AI to take.
+
+Each node in the tree executes logic and returns one of the following states as a result:
+
+- **Running** -> The task represented by this node's branch hasn't yet completed.
+- **Success** -> The node has completed its task successfully.
+- **Failure** -> The node has failed its task. This doesn't necessarily mean an error occurred. For example, a heal action might fail if the character doesn't have any potions.
+
+Different types of nodes have different responsibilities and react differently to the states of other nodes.
+
+- **Root Node**: The beginning of the tree, where the AI starts its decision-making.
+  Its job is simply to execute whatever node is below it in the tree and return its state.
+- **Sequence Node**: This node has one or more children and acts as a logical AND.
+  It executes its child nodes one at a time, and if any of them fail then the Sequence Node immediately fails.
+- **Selector Node**: This node also has one or more children, but acts as a logical OR.
+  It executes each of its children and if one of them succeeds, then the Selector Node immediately succeeds.
```

Comment:
Can we add more clarity around what `success` and `failure` mean in this context? Maybe a little example

Reply:
Added a practical example to this section